common/tools/dev-tool/src/commands/run/extract-api.ts

```diff
@@ -221,28 +219,52 @@ export default leafCommand(commandInfo, async (options) => {
     // normal extraction
     succeed = extractApi(extractorConfigObject, apiExtractorJsonPath, packageJsonPath);
   }
-
-  // Add api.json files to zip archive
-  const unscopedPackageName = projectInfo.name.split("/")[1];
   const reportTempDir = path.join(projectInfo.path, "temp");
-  const files = (await readdir(reportTempDir)).filter((f) => f.endsWith("api.json"));
-  const output = createWriteStream(path.join(reportTempDir, `${unscopedPackageName}.zip`));
-  const zip = archiver("zip");
+  const unscopedPackageName = projectInfo.name.split("/")[1];
 
-  zip.on("warning", function (err) {
-    throw err;
-  });
-  zip.on("error", function (err) {
-    throw err;
-  });
+  await buildMergedApiJson(
+    unscopedPackageName,
+    reportTempDir,
+    exports,
+    packageJson["dependencies"],
+  );
 
-  zip.pipe(output);
+  return succeed;
+});
+
+/**
+ *
+ * @returns the full path of -augmented.api.json file.
+ */
+async function buildMergedApiJson(
+  unscopedPackageName: string,
+  reportTempDir: string,
+  exports: ExportEntry[] | undefined,
+  dependencies: Record<string, string>,
+) {
+  const mainApiJsonPath = path.join(reportTempDir, `${unscopedPackageName}.api.json`);
 
-  for (const file of files) {
-    log.debug(`adding ${file} to zip archive`);
-    zip.append(createReadStream(path.join(reportTempDir, file)), { name: file });
+  const apiJson = await loadApiJson(mainApiJsonPath);
+  apiJson.metadata.dependencies = dependencies;
+  for (const subpath of exports?.filter((p) => p.baseName !== ".") ?? []) {
+    log.debug(`loading api package for "${subpath.baseName}"`);
+    const p = path.join(reportTempDir, `${unscopedPackageName}-${subpath.baseName}.api.json`);
+    try {
+      await stat(p);
```

Comment:
Why not just use `existsSync`?

Reply:
That is better! I remember some recommendation, but just double checked: only the version with callback `fs.exists` is deprecated. `existsSync` is still good.

Reply:
Updated.